src/backend/CassandraBackend.h

```diff
@@ -72,18 +72,28 @@ class BasicCassandraBackend : public BackendInterface
         if (auto const res = handle_.connect(); not res)
             throw std::runtime_error("Could not connect to Cassandra: " + res.error());
 
-        if (auto const res = handle_.execute(schema_.createKeyspace); not res)
+        if (!readOnly)
         {
-            // on datastax, creation of keyspaces can be configured to only be done thru the admin interface.
-            // this does not mean that the keyspace does not already exist tho.
-            if (res.error().code() != CASS_ERROR_SERVER_UNAUTHORIZED)
-                throw std::runtime_error("Could not create keyspace: " + res.error());
-        }
-
-        if (auto const res = handle_.executeEach(schema_.createSchema); not res)
-            throw std::runtime_error("Could not create schema: " + res.error());
+            if (auto const res = handle_.execute(schema_.createKeyspace); not res)
+            {
+                // on datastax, creation of keyspaces can be configured to only be done thru the admin interface.
+                // this does not mean that the keyspace does not already exist tho.
+                if (res.error().code() != CASS_ERROR_SERVER_UNAUTHORIZED)
+                    throw std::runtime_error("Could not create keyspace: " + res.error());
+            }
 
-        schema_.prepareStatements(handle_);
+            if (auto const res = handle_.executeEach(schema_.createSchema); not res)
+                throw std::runtime_error("Could not create schema: " + res.error());
+        }
+        try
+        {
+            schema_.prepareStatements(handle_);
+        }
+        catch (std::exception const& ex)
+        {
+            log_.error() << "Failed to prepare the statements for Clio-Readonly:" << readOnly;
```

Comment:
Maybe something like `"Failed to prepare statements: " << ex.what() << "; readonly: " << readOnly` would be slightly nicer?

Reply:
This is better but still unreadable `Failed to prepare the statements: Invalid query: keyspace clio does not exist readOnly: true`.
Please separate the `readOnly: true` part from the exception message (my recommendation is to use `;`) otherwise it looks like part of the same message 👍